### PR TITLE
Fixes #15062: store run agent batch may never catch up on very loaded system because of inefficient index and the way it handles reports

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/15062

### CODE REVIEW

**Comment by ncharles** [comment]
Commit modified

**Comment by fanf** [comment]
OK, merging this PR

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ReportsJdbcRepository.scala [hunk 1]
```diff
@@ -280,6 +280,26 @@ class ReportsJdbcRepository(doobie: Doobie) extends ReportsRepository with Logga
...
+    }
+  }
+
+  /**
+    *  utilitary methods
+    */
+
+  // Get max ID before a datetime
+  def getMaxIdBeforeDateTime(fromId: Long, before: DateTime): Box[Long] = {
+    (query[Long](s"select max(id) as id from RudderSysEvents where where id > ${fromId} and executionTimeStamp <  '${new Timestamp(before.getMillis)}''").
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ReportsJdbcRepository.scala]: extra ' here

Comment (ncharles) [reply webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ReportsJdbcRepository.scala]: extra where also

**Pull request merged**